lib/rihanna/job.ex

```diff
@@ -302,9 +309,12 @@ defmodule Rihanna.Job do
     result.rows
   end
 
-  def delete(job_id) do
+  def delete(job_id, opts \\ %{}) do
+    connection = opts[:producer_postgres_connection]
+
     result =
       producer_query(
+        connection,
```

Comment:
I'd suggest creating an `adapter()` function that takes opts and returns the connection or a default.